temporalio/converter.py

```diff
@@ -28,6 +28,7 @@
 import google.protobuf.json_format
 import google.protobuf.message
 import google.protobuf.symbol_database
+from dateutil import parser
```

Comment:
This is a new dependency just for this feature. Is there any way we can avoid it? Is there any way with the standard library to do this, and/or is the parser easy to vendor/copy into here?

(also, a new dependency would need to be added to `project.toml` which you can do via `poetry add dateutil` and then re-alphabetizing)

Reply:
Looks like it may be unavoidable to add this dependency unfortunately. Can't easily vendor stdlib impl as it's in C (ref https://github.com/python/cpython/pull/92177). But can we put a `if if sys.version_info >= (3, 11):` here to use the stdlib version? (also don't forget to add that dependency and to run `poe format`).

Also, this will need to land after (and merge main after) #172 is merged which will have Python 3.11 support.

Reply:
Not an issue on my side, I must have been working on a tainted env because I thought `dateutil` was in standard. Will rebuild and double-check.

For those wondering about why this is an issue:
https://stackoverflow.com/questions/127803/how-do-i-parse-an-iso-8601-formatted-date
https://stackoverflow.com/questions/969285/how-do-i-translate-an-iso-8601-datetime-string-into-a-python-datetime-object

On doing it within the standard library, we could hack around with the following:
```
# Example lookup for Z
map_datetime = {
    "Z": "+00:00"
}

# ...

if v[-1].isalpha() and len(v) == 20:
    v = v[:-1] + map_datetime[v[-1].upper()]

# Code as normal
```

The only reason I avoided this was to rely on a battle-tested date parser that could potentially catch a wider range of formats coming in from other languages. If you're confident we'll only see letters coming in (outside of the +/- hours format) then I'm happy to switch over to an explicit remap of the last digits for the sake of avoiding the external dependency

Reply:
I think `dateutil` is probably an acceptable solution since it doesn't seem to carry dependencies on its own or move much.

Reply:
Roger that, will add in the version check and wait for the 3.11 updates you guys have planned